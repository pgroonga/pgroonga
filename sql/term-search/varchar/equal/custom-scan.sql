CREATE TABLE tags (
  id int,
  name varchar
);

INSERT INTO tags VALUES (1, 'PostgreSQL');
INSERT INTO tags VALUES (2, 'Groonga');
INSERT INTO tags VALUES (3, 'groonga');
INSERT INTO tags VALUES (4, 'PGroonga');

CREATE INDEX pgrn_index ON tags
  USING pgroonga (name pgroonga_varchar_term_search_ops_v2);

SET pgroonga.enable_custom_scan = on;

EXPLAIN (COSTS OFF)
SELECT name
  FROM tags
 WHERE name &= 'groonga'
 ORDER BY id;

SELECT name
  FROM tags
 WHERE name &= 'groonga'
 ORDER BY id;

DROP TABLE tags;
