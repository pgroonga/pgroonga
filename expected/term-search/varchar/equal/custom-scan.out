CREATE TABLE tags (
  id int,
  name varchar
);
INSERT INTO tags VALUES (1, 'PostgreSQL');
INSERT INTO tags VALUES (2, 'Groonga');
INSERT INTO tags VALUES (3, 'groonga');
INSERT INTO tags VALUES (4, 'PGroonga');
CREATE INDEX pgrn_index ON tags
  USING pgroonga (name pgroonga_varchar_term_search_ops_v2);
SET pgroonga.enable_custom_scan = on;
EXPLAIN (COSTS OFF)
SELECT name
  FROM tags
 WHERE name &= 'groonga'
 ORDER BY id;
                       QUERY PLAN                       
--------------------------------------------------------
 Sort
   Sort Key: id
   ->  Custom Scan (PGroongaScan) on tags
         Filter: (name &= 'groonga'::character varying)
(4 rows)

SELECT name
  FROM tags
 WHERE name &= 'groonga'
 ORDER BY id;
  name   
---------
 Groonga
 groonga
(2 rows)

DROP TABLE tags;
